core/src/main/java/org/apache/iceberg/MetricsConfig.java

```diff
@@ -70,10 +71,17 @@ public static MetricsConfig fromProperties(Map<String, String> props) {
           }
           spec.columnModes.put(columnAlias, mode);
         });
-
     return spec;
   }
 
+  public void validateProperties(Schema schema) {
+    for (String column : columnModes.keySet()) {
+      if (schema.findField(column) == null) {
+        throw new ValidationException("Could not find column %s from metrics in schema %s", column, schema);
+      }
```

Comment:
You can use `ValidationException.check` like a `Precondition` to avoid adding lots of `if` statements:

```java
ValidationException.check(schema.findField(column) != null, "Could not find ...", ...);
```

Reply:
Oh nice :)